### PR TITLE
Support for principalDiagonal function #91

### PR DESCRIPTION
**Pull request by fuhrmanator**

### CODE REVIEW

**Comment by nicolas-cellier-aka-nice** [comment]
Note that Matlab `diag` is well defined for non square matrix (it always starts at upper left for diag index = 0).

It also has the ability to retrieve a sub-diagonal (diag index < 0) or a super diagonal (diag index > 0).

A Matlab-like operation is implemented in Smallapack as `diagonalAt: diagIndex`.

File: src/Math-Matrix/PMMatrix.class.st [hunk 1]
```diff
@@ -588,6 +588,17 @@ PMMatrix >> orthogonalize [
 
 ]
 
+{ #category : #'as yet unclassified' }
+PMMatrix >> principalDiagonal [
+	"https://en.wikipedia.org/wiki/Diagonal#Matrices for definitions"
+	| diag |
+	"Check for square"
```
Comment (hemalvarambhia) [inline src/Math-Matrix/PMMatrix.class.st]: I would say that the comment is redundant as the code is intention revealing enough.

File: src/Math-Matrix/PMMatrix.class.st [hunk 2]
```diff
@@ -588,6 +588,17 @@ PMMatrix >> orthogonalize [
...
+	self isSquare ifFalse: [ self error: 'Diagonal is not defined for a matrix that is not square.' ].
+	diag := PMVector new: self rows size.
```
Comment (hemalvarambhia) [inline src/Math-Matrix/PMMatrix.class.st]: Here you could use the `numberOfColumns` message (pretty sure that exists) to be a little more direct. What do you think?

Comment (fuhrmanator) [reply src/Math-Matrix/PMMatrix.class.st]: I agree with both your comments. However, I'm not sure when I'll have time to make these changes in an image and re-commit them.

**Comment by SergeStinckwich** [comment]
I think we can accept and Hemal you could do a modification after if you want.

**Comment by hemalvarambhia** [comment]
I can make the modification in a later PR, sure. Happy for this to be merged otherwise.

File: src/Math-Matrix/PMMatrix.class.st (hunk 2 above)
Comment (hemalvarambhia) [reply src/Math-Matrix/PMMatrix.class.st]: I can make the changes in another PR, no problem at all.

**Comment by SergeStinckwich** [comment]
Thank you Christopher !

**Pull request merged**